homeassistant/components/incharge/sensor.py

```diff
@@ -0,0 +1,139 @@
+"""Get status for all stations for InCharge."""
+from __future__ import annotations
+
+import datetime
+import logging
+
+import async_timeout
+from incharge.api import InCharge
+import requests
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfEnergy
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = datetime.timedelta(minutes=5)
+
+SENSOR_TYPES: list[SensorEntityDescription] = [
+    SensorEntityDescription(
+        key="total_energy_consumption",
+        name="Total Energy Consumption",
+        suggested_display_precision=2,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    )
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the InCharge sensor."""
+    # Add entities for each charging station
+    incharge_data_coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    response = await hass.async_add_executor_job(
+        incharge_data_coordinator.incharge_api.get_stations
+    )
+    for station in response.json()["stations"]:
+        entities = [
+            InChargeEntity(
+                coordinator=incharge_data_coordinator,
+                station_id=station["name"],
+                unique_id=f"{station['name']}-{description.name}",
```

Comment:
This would not appear to me as a unique_id to use names?
Why not use the username from the config entry and make that properly unique (use a unique id in the config flow)